kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/KafkaExceptionIntegrationTest.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.homeoffice.digital.sas.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_STOPPING_CONSUMING;
+
+import java.time.LocalDateTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.TestPropertySource;
+import uk.gov.homeoffice.digital.sas.config.TestConfigWithJpa;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumerServiceImpl;
+import uk.gov.homeoffice.digital.sas.model.Profile;
+import uk.gov.homeoffice.digital.sas.repository.ProfileRepository;
+import uk.gov.homeoffice.digital.sas.utils.TestUtils;
+
+/**
+ *This is seperated from the EmbeddedKafkaIntegrationTest class, due to the difficulty of overriding
+ * the schemaVersion property in an individual test.
+ *
+ */
+
+@ExtendWith(OutputCaptureExtension.class)
+@SpringBootTest(classes = TestConfigWithJpa.class)
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+@EmbeddedKafka(
+    partitions = 1
+)
+@TestPropertySource(properties = {
+    "spring.kafka.producer.bootstrap-servers=${spring.embedded.kafka.brokers}",
+    "spring.kafka.consumer.bootstrap-servers=${spring.embedded.kafka.brokers}",
+    "schemaVersion=1.0.0"})
```

Comment:
We looked at using a constant but the @TestPropertySource doesn't take constants.
1.0.0 is an invalid version because the test's are setup to validate using:
`kafka.supported.schema.version=^0.x.x`

This means any major version bump is currently not supported. The idea being that if we move a major version this will have a breaking change so we need to update accordingly.

Also side note the kafka.supported.scheam.version property can configured by the service for it's use case.